theano/tensor/nnet/corr_gemm.c

```diff
@@ -203,15 +208,20 @@ PyArrayObject* corrMM(PyArrayObject* bottom,
         output = top;
         // valid correlation: im2col, then gemm
         // Iterate over batch
+        #pragma omp parallel for schedule(static)
         for (int n = 0; n < batchSize; n++) {
+            int colidx = 0;
+#if defined(_OPENMP)
+            colidx = omp_get_thread_num();
+#endif
             // First, im2col
             im2col((%(float_type)s*)PyArray_DATA(bottom) + n * bottom_stride, nChannels, bottomHeight,
-                    bottomWidth, kH, kW, padH, padW, dH, dW, (%(float_type)s*)PyArray_DATA(col));
+                    bottomWidth, kH, kW, padH, padW, dH, dW, (%(float_type)s*)PyArray_GETPTR3(col, colidx, 0, 0));
```

Comment:
Technically `PyArray_GETPTR3` should just be doing pointer arithmetic to move past the first `colidx` matrices, so we could, although it'll make the code much less readable. In fact, looking at [the source](https://github.com/numpy/numpy/blob/master/numpy/core/include/numpy/ndarrayobject.h#L161), that's pretty much exactly what it does. [`PyArray_BYTES`](https://github.com/numpy/numpy/blob/master/numpy/core/include/numpy/ndarraytypes.h#L1438) and [`PyArray_STRIDES`](https://github.com/numpy/numpy/blob/master/numpy/core/include/numpy/ndarraytypes.h#L1450) both just do a pointer dereference, so I don't see how the problem can originate from there :/